spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkV2Filters.java

```diff
@@ -0,0 +1,356 @@
+            }
+            Expression notIn = notIn(unquote(getAttrName(childFilter.children()[0])),
+                Stream.of(notInValues).collect(Collectors.toList()));
+            return and(notNull(unquote(getAttrName(childFilter.children()[0]))), notIn);
+          } else if (hasNoInFilter(childFilter)) {
+            Expression child = convert(childFilter);
+            if (child != null) {
+              return not(child);
+            }
+          }
+          return null;
+
+        case AND: {
+          And andPredicate = (And) predicate;
+          Expression left = convert(andPredicate.left());
+          Expression right = convert(andPredicate.right());
+          if (left != null && right != null) {
+            return and(left, right);
+          }
+          return null;
+        }
+
+        case OR: {
+          Or orPredicate = (Or) predicate;
+          Expression left = convert(orPredicate.left());
+          Expression right = convert(orPredicate.right());
+          if (left != null && right != null) {
+            return or(left, right);
+          }
+          return null;
+        }
+
+        case STARTS_WITH: {
+          return startsWith(unquote(getAttrName(predicate.children()[0])),
+              convertUTF8StringIfNecessary(((LiteralValue) predicate.children()[1]).value()).toString());
+        }
+      }
+    }
+
+    return null;
+  }
+
+  private static Object convertUTF8StringIfNecessary(Object value) {
+    if (value instanceof UTF8String) {
+      return ((UTF8String) value).toString();
+    }
+    return value;
+  }
+
+  private static Expression handleEqual(String attribute, Object value) {
+    if (NaNUtil.isNaN(value)) {
+      return isNaN(attribute);
+    } else {
+      return equal(attribute, value);
+    }
+  }
+
+  private static String unquote(String attributeName) {
+    Matcher matcher = BACKTICKS_PATTERN.matcher(attributeName);
+    return matcher.replaceAll("$2");
+  }
+
+  private static boolean hasNoInFilter(Predicate predicate) {
+    Expression.Operation op = FILTERS.get(predicate.name());
+
+    if (op != null) {
+      switch (op) {
+        case AND:
+          And andPredicate = (And) predicate;
+          return hasNoInFilter(andPredicate.left()) && hasNoInFilter(andPredicate.right());
+        case OR:
+          Or orPredicate = (Or) predicate;
+          return hasNoInFilter(orPredicate.left()) && hasNoInFilter(orPredicate.right());
+        case NOT:
+          Not notPredicate = (Not) predicate;
+          return hasNoInFilter(notPredicate.child());
+        case IN:
+          return false;
+        default:
+          return true;
+      }
+    }
+
+    return false;
+  }
+
+  private static String getAttrName(org.apache.spark.sql.connector.expressions.Expression expr) {
+    return String.join(".", ((NamedReference) expr).fieldNames());
+  }
+
+  @SuppressWarnings("checkstyle:CyclomaticComplexity")
+  private static Predicate checkIfPredicateValid(Predicate predicate) {
```

Comment:
I think Spark should have this check to make sure all the V2 Filters are in valid format. Otherwise, this kind of invalid [filter](https://github.com/apache/iceberg/blob/master/spark/v3.3/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestDelete.java#L433) will get though. 

I used to have this check on Spark side, but somehow it got removed during refactor. I will add this check back. Before this is available, I will have this check here. I will remove once this is in Spark.